src/alert/internal.tsx

```diff
@@ -85,9 +82,13 @@ const InternalAlert = React.forwardRef(
       replacementContentRef,
     } = useDiscoveredContent({ type, header, children });
 
+    const [containerWidth, containerMeasureRef] = useContainerWidth();
+    const containerRef = useMergeRefs(containerMeasureRef, __internalRootRef);
     const headerRef = useMergeRefs(headerRefAction, headerRefContent);
     const contentRef = useMergeRefs(contentRefAction, contentRefContent);
 
+    const actionsRef = useActionsWrappingDetection(containerWidth, styles['action-wrapped']);
```

Comment:
Instead of solving this here, could we propagate `containerWidth` into the `ActionsWrapper` and contain offset application logic inside of it?

Bonus effect of this, we can replace `actionsRef.current.classList.add/remove` (not the best React practice) with an `isWrapped` state and conditional class in JSX

Reply:
good idea, I think that tidies it up nicely